### DISCUSSION TITLE
How to insert a payment of an invoice?

### DISCUSSION THREAD

**Discussion by Mingo07-dev** [fattureincloud/api#369] [Q&A] [answered]
After I create an invoice with the following `payments_list`, when the user needs to insert the payment should I just modify the invoice, or is there another API to use? I didn't anything like.

```
        "payments_list": [
            {
                "amount": 30,
                "due_date": "2023-01-01",
                "status": "not_paid",
                "payment_account": {
                    "id": 1127896
                }
            }
        ]
```

So then set the status to `paid `and add the `paid_date`?

**Comment by emajo** [comment, marked as answer]
Hi @Mingo07-dev, we don'have a method just to change the status of the payments, so as you said you have to make a [modify Issued Document request](https://developers.fattureincloud.it/api-reference#put-/c/-company_id-/issued_documents/-document_id-) setting the status to `paid` and add the `paid_date`.

Thanks,
Emanuele.

**Reply by Mingo07-dev** [reply]
Hi, sorry again, but I'm having some trouble with the payment. I receive this error "[422] Client error: `PUT https://api-v2.fattureincloud.it/c/1160156/issued_documents/315383860` resulted in a `422 Unprocessable Entity` response: {"error":{"message":"Invalid request.","validation_result":{"data.payments_list.0.due_date":["data.payments list.0.due d (truncated...) "

This is my code

```

    $config = FattureInCloud\Configuration::getDefaultConfiguration()->setAccessToken($_SESSION['FiCtoken']);

    $apiInstance = new FattureInCloud\Api\IssuedDocumentsApi(
    new GuzzleHttp\Client(),
    $config
    );

    $invoice = new FattureInCloud\Model\IssuedDocument;


    $invoice->setPaymentsList(
        array(
            new IssuedDocumentPaymentsListItem(
                array(
                    "status" => IssuedDocumentStatus::PAID,
                    "payment_account" => new PaymentAccount(
                        array(
                        "id" => $payment_account_id,
                        "name" => $payment_account_name
                        )
                        ),
                        "paid_date"=> $payment_day
                )
            )
        )
    );

    $get_issued_document_request = new ModifyIssuedDocumentRequest;
    $get_issued_document_request->setData($invoice);
    try {
        $result = $apiInstance->modifyIssuedDocument($company_id, $invoice_id_FIC,$get_issued_document_request);
        $fic_sent = 1;

    } catch (Exception $e) {
        $exit = $e->getMessage();
        echo ($exit);
        exit();
    }
```

**Reply by Mingo07-dev** [reply]
By printing the error and not just looking at the response I understood that in the modification I had to add the entire payment_list object, also the amount and the due_date, not just the properties I wanted to update
